src/App.jsx

```diff
@@ -1,248 +1,278 @@
-  <div className="section">
-    <div className="container">
-      <h1 className="title">Product Categories</h1>
-
-      <div className="block">
-        <nav className="panel">
-          <p className="panel-heading">Filters</p>
-
-          <p className="panel-tabs has-text-weight-bold">
-            <a
-              data-cy="FilterAllUsers"
-              href="#/"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 1
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-              className="is-active"
-            >
-              User 2
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 3
-            </a>
-          </p>
-
-          <div className="panel-block">
-            <p className="control has-icons-left has-icons-right">
-              <input
-                data-cy="SearchField"
-                type="text"
-                className="input"
-                placeholder="Search"
-                value="qwe"
-              />
-
-              <span className="icon is-left">
-                <i className="fas fa-search" aria-hidden="true" />
-              </span>
-
-              <span className="icon is-right">
-                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-                <button
-                  data-cy="ClearButton"
-                  type="button"
-                  className="delete"
-                />
-              </span>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+const productsCategories = productsFromServer.map(product => ({
+  ...product,
+  category: categoriesFromServer
+    .find(category => category.id === product.categoryId), // find by product.categoryId // find by category.ownerId
+}));
+
+const products = productsCategories.map(product => ({
+  ...product,
+  user: usersFromServer
+    .find(person => person.id === product.category.ownerId), // find by product.categoryId // find by category.ownerId
+}));
+
+export const App = () => {
+  const [currentPeople, setcurrentPeople] = useState('');
+  const [query, setQuery] = useState('');
+  const [currentCategory, setCurrentCategory] = useState('');
+  let selectedProducts = products;
+
+  if (currentPeople !== '') {
+    selectedProducts = selectedProducts
+      .filter(product => product.user.name === currentPeople);
+  }
+
+  if (query !== '') {
+    selectedProducts = selectedProducts
+      .filter(product => (
+        product.name.toLowerCase().includes(query.toLowerCase())
+      ));
+  }
+
+  if (currentCategory !== '') {
+    selectedProducts = selectedProducts
+      .filter(product => (
+        product.category.title === currentCategory
+      ));
```

Comment:
You can put all these checks into one filter method. Also, would be great to create a function for filtering and call it if query or some other filter changes